local_planner/src/nodes/waypoint_generator.cpp

```diff
@@ -69,6 +69,10 @@ void WaypointGenerator::calculateWaypoint() {
     }
   }
   last_wp_type_ = planner_info_.waypoint_type;
+  last_time_ = ros::Time::now();
```

Comment:
This should use the same timestamp as used in the current timestep (`now` in `getPathMsg`), which means that `now` should be set in here, and then passed as argument to `getPathMsg`.
This makes sure there is no time slip due to computation time in between.